src/gsi/read_gnssrspd.f90

```diff
@@ -0,0 +1,514 @@
+
+!          Convert obs lat/lon to rotated coordinate and check 
+!          if the obs is outside of the regional domain
+           if (regional) then
+              call tll2xy(dlon_earth,dlat_earth,dlon,dlat,outside)
+              if (diagnostic_reg) then
+                 call txy2ll(dlon,dlat,rlon00,rlat00)
+                 ntest      = ntest+1
+                 cdist      = sin(dlat_earth)*sin(rlat00)+cos(dlat_earth)*cos(rlat00)* &
+                             (sin(dlon_earth)*sin(rlon00)+cos(dlon_earth)*cos(rlon00))
+                 cdist      = max(-one,min(cdist,one))
+                 disterr    = acos(cdist)*rad2deg
+                 disterrmax = max(disterrmax,disterr)
+              end if
+              if(outside) cycle loop_readsb2
+           else
+              dlon = dlon_earth
+              dlat = dlat_earth
+              call grdcrd1(dlat,rlats,nlat,1)
+              call grdcrd1(dlon,rlons,nlon,1)
+           endif
+
+
+!          Read surface wind speed [m/s] from GNSSRSPD 
+           if (lspdob) then
+              !usage=r100 
+              !usage = zero  ! will be considered for assimilation
+
+              ! Get Wind Speed observations from bufr file
+              call ufbint(lunin,gnssrw,1,1,nlv,wndstr)
+              spdob = gnssrw(1,1) ! surface wind speed 
+
+              ! Don't permit observations with ws <= 1 m/s
+              if (spdob <= 1.0) cycle loop_readsb2
+              if (spdob >= missing) cycle loop_readsb2
+
+              ! Get observation error from bufr file
+              call ufbint(lunin,gnssrw,1,1,nlv,oestr)
+              obserr = max(gnssrw(1,1),1.5) ! surface wind speed error
+           endif
+
+
+           if ( .not. twodvar_regional) then
+              call deter_sfc_type(dlat_earth,dlon_earth,t4dv,isflg,tsavg)
+           endif
+
+           ! Get information from surface file necessary for conventional data
+           call deter_sfc2(dlat_earth,dlon_earth,t4dv,idomsfc,tsavg,ff10,sfcr,zz)                                                                      
+           ! Process data passed quality control 
+           igood = igood + 1
+           ndata = ndata + 1
+           nodata = nodata + 1
+           iout = ndata
+
+!-------------------------------------------------------------------------------------------------          
+
+           ! Winds --- surface wind speed 
+           if (lspdob) then
+              woe = obserr
+              !if (inflate_error) woe = woe*r3_0
+              !if (inflate_error) woe = woe*r1_2
+              !if (qcm > lim_qm ) woe = woe*1.0e6_r_kind
+              cdata_all( 1,iout)=woe                    ! wind error
+              cdata_all( 2,iout)=dlon                   ! grid relative longitude             
+              cdata_all( 3,iout)=dlat                   ! grid relative latitude                  
+              cdata_all( 4,iout)=dlnpsob                ! ln(surface pressure in cb)
```

Comment:
`dlnpsob` is declared but not defined (no given value)

Reply:
@emilyhcliu: this line was left in case some type of pressure information gets included in the BUFR files in the future. I can remove it if you think it's best.

Reply:
If the line `cdata_all( 4,iout)=dlnpsob` is retained, we must provide a default value for `dlnpsob`.  Otherwise `cdata_all(4,iout)` will be loaded with an uninitialized value.  This is unsafe coding.

Reply:
Done!